### DISCUSSION TITLE
remote address of the md file

### DISCUSSION THREAD

**Discussion by MoLing-Dong** [nuxt/content#2107] [Q&A] [answered]
Does content support the resolution of md files for remote addresses

**Comment by MoLing-Dong** [comment]
Request: Hope to use nuxt plug-in to parse the MD-format document of network request response.
Question: This project can only parse the local md file, how to parse the md file of the network request response?

**Reply by oneminch** [reply]
You can set the `sources` value in your `nuxt.config.js` to add additional data sources. Nuxt parses everything for you like its a local folder or Markdown. You can even extend its parser capabilities for Nuxt Content.

An example of this can be to use your GitHub repo that contains a list of markdown files:
```js
// nuxt.config.js
export default defineNuxtConfig({
  // ...
  content: {
    sources: {
      github: {
        prefix: '/blog', // Prefix for routes used to query contents
        driver: 'github', // Driver used to fetch contents
        repo: "mdn/content",
        branch: "main",
        dir: "files/en-us/glossary", // Directory where contents are located. It could be a subdirectory of the repository.
      },
    }
  }
});
```
The `driver` here doesn't necessarily have to be `github`. There are other supported drivers like HTTP, Local Storage, MongoDB, and Redis. You can find the configuration for each driver in the docs for [unstorage](https://unstorage.unjs.io/).

In your component, you can then do something like this:

```vue
<!-- ~/pages/Blog.vue -->
<template>
  <section>
    <a
      v-for="(post, index) in posts"
      :key="index"
      :title="post.title"
      :href="post._path"
    >{{ post.title }}</a>
  </section>
</template>

<script setup>
const { data: posts } = await useAsyncData("blog", () =>
    queryContent("/").find()
);
</script>
```

For more info, refer [the documentation for `sources`](https://content.nuxtjs.org/api/configuration#sources) and [the documentaion for unstorage](https://unstorage.unjs.io/).

I hope that helped!

**Reply by MoLing-Dong** [reply]
Thank you very much, but I still get the content of the index.md file with local content by using this method. I don't know why, but I use the code you gave me

**Reply by oneminch** [reply]
Can you share the contents of your `nuxt.config` file?

**Reply by MoLing-Dong** [reply]
![image](https://github.com/nuxt/content/assets/70124717/ca146c47-b3e3-40a0-8ecd-73ec9135af19)

**Reply by oneminch** [reply]
Are you accessing the files from a `/blog` prefix like for instance `localhost:3000/blog/mvc`?

**Reply by MoLing-Dong** [reply]
That doesn't seem to work. I got a 404

**Reply by MoLing-Dong** [reply]
I want to show my lot in my project readme, https://github.com/MoLing-Dong/MoLing-Dong/blob/main/README.md, up to now I didn't understand how should I set

**Reply by oneminch** [reply]
In that case you need to modify your nuxt config to parse that repo. The config you sent me parses `nuxt/content`. The `repo` property in your `sources.github` needs to be `MoLing-Dong/MoLing-Dong` and `dir` needs to be `/`. I think you'll then be able to access your file at `localhost:3000/blog/readme`

**Reply by oneminch** [reply, marked as answer]
You should also have a `[...slug].vue` file with the following code as your catch-all route. 

```vue
<template>
  <main>
    <ContentDoc />
  </main>
</template>
```

I recommend you read more about it [here](https://content.nuxtjs.org/guide/displaying/rendering/)

**Reply by MoLing-Dong** [reply]
![image](https://github.com/nuxt/content/assets/70124717/b2a6df3a-3a5f-4fd3-85fb-bc67b5451f44)
![image](https://github.com/nuxt/content/assets/70124717/2c8e05c3-6793-42ce-8cbf-0d49ac856807)

**Reply by oneminch** [reply]
Any progress?

**Reply by MoLing-Dong** [reply]
I finished, but I still encountered this question leg when using it
![image](https://github.com/nuxt/content/assets/70124717/a3ef9838-02fc-4820-9dc4-52a5fd65fbd7)

**Reply by nobkd** [reply]
That currently seems to be an issue with code blocks.
Also tracked in https://github.com/nuxt/content/issues/2111

**Reply by MoLing-Dong** [reply]
I think I should wait for content to grow up

**Reply by oneminch** [reply]
If it has been a while since you started your project, try updating your npm packages (`npm update`). I found that to be helpful at times. You may never know a fix has already been issued in a newer version.

This might also be an issue with the UI library which I think is Element Plus. Experiment with using a different element like a `<div>` just to determine if the issue is from there.

> **Note** 
> Also note that if you named your page file `[...slug].vue` (a catch-all route), routes that start with `/blog` like `/blog/anything-here` and **possibly all your pages** might display your `/blog/readme` page. 

I used Nuxt content on my portfolio to display my blog content from GitHub which contains code blocks like yours. And it worked perfectly. You can check it out at https://oneminch.dev/blog. The code is also open-source at https://github.com/oneminch/portfolio if you want to inspect it.
